File: ruby/009/main.rb
=begin
    ProjectEuler 009
    Ruby / main.rb
    copyright (c) 2015 Susisu

    Ruby 2.0.0p481
    `ruby main.rb`
=end

def main
    a = 1
    while a <= 332
        b = a + 1
        while b < 1000 - a - b
            if 2 * a * b - 2000 * (a + b) + 1000000 == 0 then
                puts a * b * (1000 - a - b)
                return
            end
            b += 1
        end
        a += 1
    end
end

main()

File: julia/009/main.jl
# ProjectEuler 009
# Julia / main.jl
# copyright (c) 2015 Susisu
#
# Julia v0.4.1
# `julia main.jl`

function main()
    for a = 1:332
        for b = a:div((1000 - a), 2)
            c = 1000 - a - b
            if a * a + b * b == c * c
                println(a * b * c)
                return
            end
        end
    end
end

main()
